controllers/backstage_controller_test.go

```diff
@@ -1445,6 +1445,39 @@ plugins: []
 				},
 			)
 		})
+
+		When("setting invalid replicas", func() {
```

Comment:
What are we testing by this 30 lines of code? 
this code:
deployment.Spec.Replicas = backstage.Spec.Application.Replicas
or
Kubernete's ability to process a negative amount of replicas? :)

Reply:
The purpose is that when an invalid configuration is applied an error message should be logged in the CR status.

Reply:
Well, I can see:
When("setting invalid replicas"
It("Should fail to reconsile...

BTW, if it is just to check the status string could you please consider checking it in some else testcase instead (it is already too overloaded as you can see).

Reply:
Do you mean to move it into part of another test case? Thanks.

Reply:
Not exactly, I rather mean if the purpose is to check if the status is set correctly, whatever the reason of failing is, you probably can reuse some other "should fail to reconcile..." case instead...

Well, my point is that testing negative replicas number (along with status printing) you IMO rather test k8s than backstage operator.
Maybe it makes sense to find something other to test or I missed something?

Reply:
The intent of this test was not to test k8s itself, and it is used to verify that an error can appear in the status, regardless it is caused by the operator itself or caused by the underlying k8s. So I think it should be fine here.